src/main/java/com/minecolonies/core/datalistener/DiseasesListener.java

```diff
@@ -0,0 +1,234 @@
+                {
+                    cureString.append(" + ");
+                }
+            }
+            return cureString;
+        }
+    }
+
+    /**
+     * Default constructor.
+     */
+    public DiseasesListener()
+    {
+        super(GSON, "colony/diseases");
+    }
+
+    /**
+     * Sync to client.
+     *
+     * @param player to send it to.
+     */
+    public static void sendGlobalDiseasesPackets(final ServerPlayer player)
+    {
+        final FriendlyByteBuf byteBuf = new FriendlyByteBuf(Unpooled.buffer());
+        byteBuf.writeInt(DISEASES.getAll().size());
+        for (final Disease disease : DISEASES.getAll())
+        {
+            byteBuf.writeResourceLocation(disease.id);
+            byteBuf.writeComponent(disease.name);
+            byteBuf.writeInt(disease.rarity);
+            for (final ItemStorage cureItem : disease.cureItems)
+            {
+                StandardFactoryController.getInstance().serialize(byteBuf, cureItem);
+            }
+        }
+        Network.getNetwork().sendToPlayer(new GlobalDiseaseSyncMessage(byteBuf), player);
+    }
+
+    /**
+     * Read the data from the packet and parse it.
+     *
+     * @param byteBuf pck.
+     */
+    public static void readGlobalDiseasesPackets(final FriendlyByteBuf byteBuf)
+    {
+        final RandomCollection<ResourceLocation, Disease> newDiseases = new RandomCollection<>();
+        final int size = byteBuf.readInt();
+        for (int i = 0; i < size; i++)
+        {
+            final ResourceLocation id = byteBuf.readResourceLocation();
+            final Component name = byteBuf.readComponent();
+            final int rarity = byteBuf.readInt();
+
+            final List<ItemStorage> cureItems = new ArrayList<>();
+            final int itemCount = byteBuf.readInt();
+            for (int j = 0; j < itemCount; j++)
+            {
+                cureItems.add(StandardFactoryController.getInstance().deserialize(byteBuf));
+            }
+
+            newDiseases.add(rarity, id, new Disease(id, name, rarity, cureItems));
+        }
+        DISEASES = newDiseases;
+    }
+
+    /**
+     * Get a collection of all possible diseases.
+     *
+     * @return the collection of diseases.
+     */
+    @NotNull
+    public static Collection<Disease> getDiseases()
+    {
+        return DISEASES.getAll();
+    }
+
+    /**
+     * Get a specific disease by id.
+     *
+     * @param id the disease id.
+     * @return the disease instance or null if it does not exist.
+     */
+    @Nullable
+    public static Disease getDisease(final ResourceLocation id)
+    {
+        return DISEASES.get(id);
+    }
+
+    /**
+     * Get a random disease from the list of diseases.
+     *
+     * @param random the random provider.
+     * @return the random disease instance or null if no diseases exist.
+     */
+    @Nullable
+    public static Disease getRandomDisease(final RandomSource random)
+    {
+        return DISEASES.next(random);
```

Comment:
can just use mathutils random here and make this a parameterless method

Reply:
The random collection was built with Mojang their random source in mind, not the regular random instance. So that does not work. Plus this is a non issue because this is called from the citizen, so I already have access to a random with 0 effort.

Reply:
Ah, then we should probably adjust this to just a normal random. Makes it easier to re-use elsewhere and easier to port when mojang does randomsource adjustments. Like this just needs a normal random, doesn't it? The citizen randoms can cause issues at time, so I prefer avoid using them unless impossible otherwise.